BPR.Tests/AnalysisServiceTests.cs

```diff
@@ -36,123 +39,197 @@ public class AnalysisServiceTests {
+        {
+            new()
+            {
                 Using = "using BPR.Persistence",
-                File = "DependencyController.cs"
+                ComponentName = "BPRBlazor"
+            }
+        };
+    }
+
+    private static List<UsingDirective> GenerateDummySelfReferencingUsingDirectives()
+    {
+        return new List<UsingDirective>
+        {
+            new()
+            {
+                Using = "using BPRBlazor",
+                ComponentName = "BPRBlazor"
             }
         };
-        var component = new AnalysisArchitecturalComponent() {
-            Name = "Component X",
-            NamespaceComponents = new List<AnalysisNamespace>() {
-                new() {
-                    Name = "BPR/Analysis/Services"
+    }
+
+    private static AnalysisArchitecturalComponent GenerateDummyComponent()
+    {
+        return new AnalysisArchitecturalComponent
+        {
+            Name = "1",
+            NamespaceComponents = new()
+            {
+                new()
+                {
+                    Name = "BPRBlazor"
                 }
             },
-            Dependencies = new List<AnalysisArchitecturalComponent>() {
-                new (){
-                    Name = "Component Y",
-                    NamespaceComponents = new List<AnalysisNamespace>() {
-                        new () {
-                            Name = "BPR/Analysis/Models"
+            Dependencies = new()
+            {
+                new()
+                {
+                    Name = "2",
+                    NamespaceComponents = new()
+                    {
+                        new()
+                        {
+                            Name = "BPR.Mediator"
+                        }
+                    },
+                    Dependencies = new()
+                    {
+                        new()
+                        {
+                            Name = "3",
+                            NamespaceComponents = new()
+                            {
+                                new()
+                                {
+                                    Name = "BPR.Persistence"
+                                }
+                            }
                         }
                     }
-                    
                 }
             }
         };
+    }
 
-        //Act
-        var result = AnalysisService.GetDependencyAnalysisOnComponent(usingList, component);
-        foreach (var dependency in component.Dependencies) {
-            result.AddRange(AnalysisService.GetDependencyAnalysisOnComponent(usingListDependency,dependency));
-        }
-        //Assert
-        Assert.That(result, !Is.Empty);
-        Assert.That(result[0].Severity,Is.EqualTo(ViolationSeverity.Major));
-        Assert.That(result[0].Type,Is.EqualTo(ViolationType.ForbiddenDependency));
+    private static AnalysisArchitecturalComponent GenerateDummyComponentWithNoDependencies()
+    {
```

Comment:
would it make sense to have this separated from the test in another class?

Reply:
Yeah, will do so.